docs/content/integrations/bigquery/using-bigquery-with-dagster.mdx

```diff
@@ -34,18 +47,195 @@ To complete this tutorial, you'll need:
+You can also specify a `location` where computation should take place.
+
+```python file=/integrations/bigquery/tutorial/resource/configuration.py startafter=start_example endbefore=end_example
+from dagster_gcp import BigQueryResource
+
+from dagster import Definitions
+
+defs = Definitions(
+    assets=[iris_data],
+    resources={
+        "bigquery": BigQueryResource(
+            project="my-gcp-project",  # required
+            location="us-east5",  # optional, defaults to the default location for the project - see https://cloud.google.com/bigquery/docs/locations for a list of locations
+        )
+    },
+)
+```
+
+### Step 2: Create tables in BigQuery
+
+<TabGroup>
+
+<TabItem name="Create BigQuery tables in Dagster">
+
+#### Create BigQuery tables in Dagster
+
+Using the BigQuery resource, you can create BigQuery tables using the BigQuery Python API:
+
+```python file=/integrations/bigquery/tutorial/resource/create_table.py startafter=start_example endbefore=end_example
+import pandas as pd
+from dagster_gcp import BigQueryResource
+
+from dagster import asset
+
+
+@asset
+def iris_data(bigquery: BigQueryResource) -> None:
+    iris_df = pd.read_csv(
+        "https://docs.dagster.io/assets/iris.csv",
+        names=[
+            "sepal_length_cm",
+            "sepal_width_cm",
+            "petal_length_cm",
+            "petal_width_cm",
+            "species",
+        ],
+    )
+
+    with bigquery.get_client() as client:
+        job = client.load_table_from_dataframe(
+            dataframe=iris_df,
+            destination="iris.iris_data",
+        )
+        job.result()
+```
+
+In this example, you're defining an asset that fetches the Iris dataset as a Pandas DataFrame and renames the columns. Then, using the BigQuery resource, the DataFrame is stored in BigQuery as the `iris.iris_data` table.
+
+Now you can run `dagster dev` and materialize the `iris_data` asset from the Dagster UI.
+
+</TabItem>
+
+<TabItem name="Make existing tables available in Dagster">
+
+#### Make existing tables available in Dagster
+
+If you already have tables in BigQuery, you may want to have other assets in Dagster depend on those tables. You can accomplish this by creating [source assets](/concepts/assets/software-defined-assets#defining-external-asset-dependencies) for these tables.
+
+```python file=/integrations/bigquery/tutorial/resource/source_asset.py
+from dagster import SourceAsset
+
+iris_harvest_data = SourceAsset(key="iris_harvest_data")
+```
+
+In this example, you're creating a <PyObject object="SourceAsset" /> for a pre-existing table called `iris_harvest_data`.
+
+</TabItem>
+
+</TabGroup>
+
+### Step 3: Define downstream assets
+
+Once you have created an asset or source asset that represents a table in BigQuery, you will likely want to create additional assets that work with the data.
```

Comment:
i feel 
```suggestion
Once you have created an asset that represents a table in BigQuery, you will likely want to create additional assets that work with the data.
```
i feel we can simply say "asset" cos my mental model is source asset is a special asset.